docs/spec/BeakerSpec.tex

```diff
@@ -0,0 +1,994 @@
+This is a critical feature. \emph{All} data stored by the kernel is stored in a
+region of \emph{storage} called \emph{kernel storage}. \emph{No} other data can
+be stored in \emph{kernel storage}.
+
+This kernel storage is defined as all storage locations within a certain storage
+range. All values in kernel storage share the same storage key prefix. This
+prefix is currently 0x00, and so currently includes all values in the interval
+$[0\text{x}0000 \cdots 0000, 0\text{x}0100 \cdots 0000)$ (some zeros have been
+elided for typographic reasons, but each values is 32 bytes).
+
+\begin{table}[H]
+  \caption{Overview of kernel storage layout.}
+  \centering{}%
+  \begin{tabularx}{\textwidth}{c | c | c | c | c | X }
+    \hline
+    \texttt{0x} & \texttt{00} & \texttt{01 - 04} & \texttt{05 - 1f} & \texttt{1e - 20} & Description \\
+    \hline
+    \hline
+    \texttt{0x} & \texttt{00} & \texttt{00 00 00 00} & Proc Key & $\cdots$ & Procedure Heap \\
+    \texttt{0x} & \texttt{00} & \texttt{00 00 00 01} & \texttt{00 $\cdots$ 00} & \texttt{00 00 00} & \# of Procedures\\
+    \texttt{0x} & \texttt{00} & \texttt{00 00 00 01} & Key Index & \texttt{00 00 00} & Procedure List \\
+    \texttt{0x} & \texttt{00} & \texttt{00 00 00 02} & \texttt{00 $\cdots$ 00} & \texttt{00 00 00} & Kernel Address \\
+    \texttt{0x} & \texttt{00} & \texttt{00 00 00 03} & \texttt{00 $\cdots$ 00} & \texttt{00 00 00} & Current Procedure \\
+    \texttt{0x} & \texttt{00} & \texttt{00 00 00 04} & \texttt{00 $\cdots$ 00} & \texttt{00 00 00} & Entry Procedure \\
+    \hline
+  \end{tabularx}
+\end{table}
+
+\section{Procedures}\label{procedures}
+Data about procedure is store in 2 locations: the procedure table, and the
+procedure heap. The procedure table is simply a managed, enumerable, and
+iterable list of procedure keys (procedure key being something that identifies a
+particular procedure). The data associated with a procedure, such as its
+Ethereum address and its capabilities are stored on the procedure heap. Given a
+procedure key, the data associated with a procedure can be located using that
+key.
+
+\subsection{Procedure Key}
+Each procedure is defined by a procedure key. This key is a sequence of 24 8-bit
+bytes. This is treated as a sequence of 192~bits in all cases.
+
+The kernel itself treats the procedure keys as an opaque 192-bit value, however,
+the capability system and some functions apply prefixes to these key value. For
+example: capabilities that rely on procedure key values will often use a prefix
+value to define a range of keys. This allows one to defined a hierarchy of
+procedures, much like a directory structrue. See those capabilities for more
+detail.
+
+\subsection{Key Index}
+Each procedure key included in the kernel is associated with a key index, which
+identifies a procedure in the procedure table. This is not used anywhere except
+the procedure table itself. It is 1-based, that is the first value is 1, and the
```

Comment:
I need help to understand this. Is this based on the current implementation?

Reply:
This is the key index, which is the procedure list index. I'll change it to *Procedure Index*, I think that's a clearer name.